pkg/operator/observe_config.go

```diff
@@ -149,6 +150,63 @@ func observeInternalRegistryHostname(kubeClient kubernetes.Interface, clientConf
 	return observedConfig, nil
 }
 
+func observeBuildControllerConfig(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+	buildConfigClient, err := configv1client.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+	build, err := buildConfigClient.Builds().Get("cluster", metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		return observedConfig, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	// Set build defaults
+	gitHTTPProxy := build.Spec.BuildDefaults.GitHTTPProxy
+	if len(gitHTTPProxy) > 0 {
+		unstructured.SetNestedField(observedConfig, gitHTTPProxy, "build", "buildDefaults", "gitHTTPProxy")
+	}
+	gitHTTPSProxy := build.Spec.BuildDefaults.GitHTTPSProxy
+	if len(gitHTTPSProxy) > 0 {
+		unstructured.SetNestedField(observedConfig, gitHTTPSProxy, "build", "buildDefaults", "gitHTTPSProxy")
+	}
+	gitNoProxy := build.Spec.BuildDefaults.GitNoProxy
+	if len(gitNoProxy) > 0 {
+		unstructured.SetNestedField(observedConfig, gitNoProxy, "build", "buildDefaults", "gitNoProxy")
+	}
+	env := build.Spec.BuildDefaults.Env
+	if len(env) > 0 {
+		unstructured.SetNestedField(observedConfig, env, "build", "buildDefaults", "env")
+	}
+	labels := build.Spec.BuildDefaults.ImageLabels
+	if len(labels) > 0 {
+		unstructured.SetNestedField(observedConfig, labels, "build", "buildDefaults", "imageLabels")
+	}
+	// Set build overrides
+	overrideLabels := build.Spec.BuildOverrides.ImageLabels
+	if len(overrideLabels) > 0 {
+		unstructured.SetNestedField(observedConfig, overrideLabels, "build", "buildOverrides", "imageLabels")
+	}
+	// TODO: make NodeSelector in controller config a corev1.LabelSelector?
+	nodeSelector := build.Spec.BuildOverrides.NodeSelector
+	if len(nodeSelector.MatchLabels) > 0 {
+		unstructured.SetNestedField(observedConfig, nodeSelector.MatchLabels, "build", "buildOverrides", "nodeSelector")
+	}
+	tolerations := build.Spec.BuildOverrides.Tolerations
+	if len(tolerations) > 0 {
+		unstructured.SetNestedField(observedConfig, tolerations, "build", "buildOverrides", "tolerations")
+	}
+	// TODO: 1) generate CA bundle from ConfigMapRef
+	//       2) write CA bundle to a ConfigMap in the controller-manager's namespace
+	//       3) wire in logic to force a rollout if CA bundle changes
```

Comment:
// (4) add a watch to watch the CA bundle configmap(pointed to by the ConfigMapRef) so that when it changes, the operator can update the bundle in the controller-manager's configmap and do a rollout.

Reply:
(and of course if the configmapref changes, the watch needs to be replaced/updated to watch the new referenced configmap)

I believe @deads2k has some utilities/patterns for doing a single resource watch that he can point you to.